src/iterators/bfs.jl

```diff
@@ -75,35 +76,33 @@ Iterator to visit vertices in a graph using breadth-first search.
 """
 function Base.iterate(t::BFSIterator, state::BFSVertexIteratorState)
     graph, visited, queue = t.graph, state.visited, state.queue
-    while !isempty(queue)
-        if state.n_visited == nv(graph)
-            return nothing
-        end
```

Comment:
Does that have any influence on performance? Because I feel like expliectly stating that we return `nothing` in that case makes it easier to understand what is going on.